pipeline/main.yaml

```diff
@@ -0,0 +1,76 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: "Main template for a CI/CD Serverless application."
+Parameters:
+  AppName:
+    Type: String
+    Description: Name of the application
+    MinLength: "1"
+    MaxLength: "80"
+    AllowedPattern: "[A-Za-z0-9-]+"
+    ConstraintDescription: Malformed input parameter. AppName must only contain upper and lower case letters, numbers, and -.
+  SAMInputFile:
+    Type: String
+    Description: The filename for the SAM file.
+    Default: slack-anonymous.yml
+  SAMOutputFile:
+    Type: String
+    Description: The filename for the output SAM file from the buildspec file.
+    Default: post-saml.yaml
+  SlackWebHookUrl:
+    Type: String
+    Description: Slack WebHook URl
+  SlackChannel:
+    Type: String
+    Description: Slack channel
+    Default: "testing1"
+  CodeBuildImage:
+    Type: String
+    Default: "aws/codebuild/nodejs:4.3.2"
+    Description: Image used for CodeBuild project.
+  CICDS3BucketName:
+    Type: String
+    Default: "anon-cicd"
+    Description: S3 Bucket where the pipeline templates live
+  GitHubRepoName:
+    Type: String
+    Description: The GitHub repo name
+  GitHubRepoBranch:
+    Type: String
+    Description: The GitHub repo branch code pipelines should watch for changes on
+    Default: master
+  GitHubUser:
+    Type: String
+    Description: GitHub UserName. This username must have access to the GitHubToken.
+  GitHubToken:
+    NoEcho: true
+    Type: String
+    Description: "Secret. OAuthToken with access to Repo. Long string of characters and digits. Go to https://github.com/settings/tokens"
+Resources:
+  ServerlessRoles:
+    Type: "AWS::CloudFormation::Stack"
+    Properties:
+      Parameters:
+        AppName: !Ref AppName
+      TemplateURL:
+        Fn::Sub: https://${CICDS3BucketName}.s3.amazonaws.com/templates/pipeline-roles.yaml
+  ServerlessPipeline:
+    Type: "AWS::CloudFormation::Stack"
+    DependsOn: [ServerlessRoles]
+    Properties:
+      Parameters:
+        AppName: !Ref AppName
+        SAMInputFile: !Ref SAMInputFile
+        SAMOutputFile: !Ref SAMOutputFile
+        SlackWebHookUrl: !Ref SlackWebHookUrl
+        SlackChannel: !Ref SlackChannel
+        CodeBuildImage: !Ref CodeBuildImage
+        GitHubRepoName: !Ref GitHubRepoName
+        GitHubRepoBranch: !Ref GitHubRepoBranch
+        GitHubUser: !Ref GitHubUser
+        GitHubToken: !Ref GitHubToken
+        CICDS3BucketName: !Ref CICDS3BucketName
+        CodePipelineRole: !GetAtt ServerlessRoles.Outputs.CodePipelineRole
+        CloudformationRole: !GetAtt ServerlessRoles.Outputs.CloudformationDeployRole
+        CodeBuildRole: !GetAtt ServerlessRoles.Outputs.CodeBuildRole
+      TemplateURL:
+        Fn::Sub: https://${CICDS3BucketName}.s3.amazonaws.com/templates/pipeline.yaml
```

Comment:
Hardcoding the `templates` prefix is not a good idea. Rather make this another parameter. For example
[Pivotal QS Link Example](https://github.com/aws-quickstart/quickstart-pivotal-cloudfoundry/blob/master/templates/pivotal-cloudfoundry.template#L306) and [PCF Param Validation Example](https://github.com/aws-quickstart/quickstart-pivotal-cloudfoundry/blob/master/templates/pivotal-cloudfoundry.template#L200-L208)

Remember to adjust this everywhere you use S3. Like the Swagger template etc..